src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java

```diff
@@ -73,29 +75,22 @@ protected Vec3d getEntityPosition()
 
     /**
      * Get the destination from the path.
-     * @return the destionation position.
+     *
+     * @return the destination position.
      */
     public BlockPos getDestination()
     {
         return destination;
     }
 
-    @Nullable
     @Override
-    public Path getPathToPos(final BlockPos pos)
+    protected boolean isDirectPathBetweenPoints(final Vec3d start, final Vec3d end, final int sizeX, final int sizeY, final int sizeZ)
     {
-        //Because this directly returns Path we can't do it async.
-        return null;
+        // TODO improve road walking. This is better in some situations, but still not great.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)